### PR TITLE
doc(upgrade): add upgrade and migration docs.

### PR DESCRIPTION
**Pull request by akhilerm**
- add upgrade docs for 2.0.0 release
- add migration section

Signed-off-by: Akhil Mohan <akhil.mohan@mayadata.io>

### CODE REVIEW

File: k8s/upgrades/README.md [hunk 1]
```diff
@@ -111,40 +120,6 @@ the OpenEBS Control Plane components.
...
-  openebs-ndm-5mzg9                              1/1     Running   0          103s
-  openebs-ndm-bmjxx                              1/1     Running   0          107s
-  openebs-ndm-operator-5ffdf76bfd-ldxvk          1/1     Running   0          115s
-  openebs-ndm-v7vd8                              1/1     Running   0          114s
-  openebs-provisioner-678c549559-gh6gm           1/1     Running   0          2m8s
-  openebs-snapshot-operator-75dc998946-xdskl     2/2     Running   0          2m6s
-  ```
-
 **Note: If you have any queries or see something unexpected, please reach out to the
 OpenEBS maintainers via [Github Issue](https://github.com/openebs/openebs/issues) or via [OpenEBS Slack](https://slack.openebs.io).**
```
Comment (shubham14bajpai) [inline k8s/upgrades/README.md:124]:
```suggestion
OpenEBS maintainers via [Github Issue](https://github.com/openebs/openebs/issues) or via #openebs channel on [Kubernetes Slack](https://slack.k8s.io).**
```

File: k8s/upgrades/1.x/README.md [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+  openebs-ndm-5mzg9                              1/1     Running   0          103s
+  openebs-ndm-bmjxx                              1/1     Running   0          107s
+  openebs-ndm-operator-5ffdf76bfd-ldxvk          1/1     Running   0          115s
+  openebs-ndm-v7vd8                              1/1     Running   0          114s
+  openebs-provisioner-678c549559-gh6gm           1/1     Running   0          2m8s
+  openebs-snapshot-operator-75dc998946-xdskl     2/2     Running   0          2m6s
+  ```
+
+**Note: If you have any queries or see something unexpected, please reach out to the
+OpenEBS maintainers via [Github Issue](https://github.com/openebs/openebs/issues) or via [OpenEBS Slack](https://slack.openebs.io).**
```
Comment (shubham14bajpai) [inline k8s/upgrades/1.x/README.md:149]:
```suggestion
OpenEBS maintainers via [Github Issue](https://github.com/openebs/openebs/issues) or via #openebs channel on [Kubernetes Slack](https://slack.k8s.io).**
```

File: k8s/upgrades/README.md [hunk 3]
```diff
@@ -412,8 +373,10 @@ spec:
...
         # the image version should be same as the --to-version mentioned above
         # in the args of the job
-        image: quay.io/openebs/m-upgrade:1.12.0
+        image: openebs/m-upgrade:2.0.0
         imagePullPolicy: Always
       restartPolicy: OnFailure
 ---
 ```
+
+### Migration of Pools and Volumes to latest CSI based Volumes
```
Comment (shubham14bajpai) [inline k8s/upgrades/README.md:382]:
```suggestion
## Migration of Pools and Volumes to latest CSI based Volumes

OpenEBS 2.0.0 moves the cStor engine towards `v1` schema and CSI based provisioning. To migrate from old SPC base pools and cStor external-provisioned volume to CSPC based pools and cStor CSI volumes follow the steps mentioned in the [doc](https://github.com/openebs/upgrade/blob/master/docs/migration.md)

## Upgrading CSPC pools and cStor CSI volumes

CSPC pools and cStor CSI volumes can be upgraded from `1.10.0` or later to the latest release via steps mentioned in the [doc](https://github.com/openebs/upgrade/blob/master/docs/upgrade.md)
```

File: k8s/upgrades/README.md [hunk 4]
```diff
@@ -67,6 +80,14 @@ machine that has access to Kubernetes master**
 Upgrade steps vary depending on the way OpenEBS was installed by you.
 Below are steps to upgrade using some common ways to install OpenEBS:
 
+### Prerequisite for control plane upgrade
+1. Make sure all the blockdevices that are in use by cstor or localPV are connected to the node.
+2. Make sure all the blockdevices that are in use are in active and claimed state.
+3. Make sure that all manually created and claimed blockdevices are excluded in the NDM configmap path
+filter.
+
+**NOTE: Upgrade of LocalPV rawblock volumes are not supported. Please exclude it in configmap**
```
Comment (kmova) [inline k8s/upgrades/README.md:89]: If any of the above pre-requisites are not met? what should the user do? Reach out support via slack? 

Also, for each of the steps, is there a kubectl command that can be added that user can run and verify that pre-requisite is met? If the instructions are long, a new pre-req.md file can be created with detailed instructions. And a link added here.

Comment (akhilerm) [reply k8s/upgrades/README.md:88]: Done the changes.

File: k8s/upgrades/README.md [hunk 5]
```diff
@@ -1,13 +1,26 @@
...
 This document describes the steps for the following OpenEBS Upgrade paths:
 
-- Upgrade from 1.0.0 or later to a newer release up to 1.12.0
+- Upgrade from 1.0.0 or later to a newer release up to 2.0.0
 
-For other upgrade paths, please refer to the respective directories.
-Example, the steps to upgrade from 0.9.0 to 1.0.0 will be under [0.9.0-1.0.0](./0.9.0-1.0.0/).
+For other upgrade paths of earlier releases, please refer to the respective directories.
+Example: 
+- the steps to upgrade from 0.9.0 to 1.0.0 will be under [0.9.0-1.0.0](./0.9.0-1.0.0/).
```
[the remark this answers is not in the document]
Comment (kmova) [reply k8s/upgrades/README.md:23]: We will potentially have some 1.12.1 and further releases as the older schema gets deprecated.

[the remark this answers is not in the document]
Comment (shubham14bajpai) [reply k8s/upgrades/README.md:23]: Added the old README.

**Pull request merged**